templates/repo/home.tmpl

```diff
@@ -4,171 +4,222 @@
-		<div class="ui form tw-hidden tw-flex tw-gap-2 tw-my-2" id="topic_edit">
-			<div class="ui fluid multiple search selection dropdown tw-flex-wrap tw-flex-1">
-				<input type="hidden" name="topics" value="{{range $i, $v := .Topics}}{{.Name}}{{if Eval $i "+" 1 "<" (len $.Topics)}},{{end}}{{end}}">
-				{{range .Topics}}
-					{{/* keep the same layout as Fomantic UI generated labels */}}
-					<a class="ui label transition visible tw-cursor-default tw-inline-block" data-value="{{.Name}}">{{.Name}}{{svg "octicon-x" 16 "delete icon"}}</a>
-				{{end}}
-				<div class="text"></div>
-			</div>
-			<div>
-				<button class="ui basic button" id="cancel_topic_edit">{{ctx.Locale.Tr "cancel"}}</button>
-				<button class="ui primary button" id="save_topic" data-link="{{.RepoLink}}/topics">{{ctx.Locale.Tr "save"}}</button>
-			</div>
-		</div>
-		{{end}}
-		{{if .Repository.IsArchived}}
-			<div class="ui warning message tw-text-center">
-				{{if .Repository.ArchivedUnix.IsZero}}
-					{{ctx.Locale.Tr "repo.archive.title"}}
-				{{else}}
-					{{ctx.Locale.Tr "repo.archive.title_date" (DateUtils.AbsoluteLong .Repository.ArchivedUnix)}}
-				{{end}}
-			</div>
-		{{end}}
-		{{template "repo/sub_menu" .}}
-		{{$n := len .TreeNames}}
-		{{$l := Eval $n "-" 1}}
-		{{$isHomepage := (eq $n 0)}}
-		<div class="repo-button-row" data-is-homepage="{{$isHomepage}}">
-			<div class="repo-button-row-left">
-				{{$branchDropdownCurrentRefType := "branch"}}
-				{{$branchDropdownCurrentRefShortName := .BranchName}}
-				{{if .IsViewTag}}
-					{{$branchDropdownCurrentRefType = "tag"}}
-					{{$branchDropdownCurrentRefShortName = .TagName}}
-				{{end}}
-				{{template "repo/branch_dropdown" dict
-					"Repository" .Repository
-					"ShowTabBranches" true
-					"ShowTabTags" true
-					"CurrentRefType" $branchDropdownCurrentRefType
-					"CurrentRefShortName" $branchDropdownCurrentRefShortName
-					"CurrentTreePath" .TreePath
-					"RefLinkTemplate" "{RepoLink}/src/{RefType}/{RefShortName}/{TreePath}"
-					"AllowCreateNewRef" .CanCreateBranch
-					"ShowViewAllRefsEntry" true
-				}}
-				{{if and .CanCompareOrPull .IsViewBranch (not .Repository.IsArchived)}}
-					{{$cmpBranch := ""}}
-					{{if ne .Repository.ID .BaseRepo.ID}}
-						{{$cmpBranch = printf "%s/%s:" (.Repository.OwnerName|PathEscape) (.Repository.Name|PathEscape)}}
-					{{end}}
-					{{$cmpBranch = print $cmpBranch (.BranchName|PathEscapeSegments)}}
-					{{$compareLink := printf "%s/compare/%s...%s" .BaseRepo.Link (.BaseRepo.DefaultBranch|PathEscapeSegments) $cmpBranch}}
-					<a id="new-pull-request" role="button" class="ui compact basic button" href="{{$compareLink}}"
-						data-tooltip-content="{{if .PullRequestCtx.Allowed}}{{ctx.Locale.Tr "repo.pulls.compare_changes"}}{{else}}{{ctx.Locale.Tr "action.compare_branch"}}{{end}}">
-						{{svg "octicon-git-pull-request"}}
-					</a>
+
+		<div class="flex-container">
+			<div class="flex-container-main" data-home={{.IsHomePage}}>
```

Comment:
Same name, different logic?

![image](https://github.com/user-attachments/assets/4d584437-d08f-4db6-94e3-e28b353026ce)

Reply:
Comes from the original PR. It is unnecessary.
Done in 8a865de6530e6cc83d7791853d66aa044f05aa05